File: backend/src/main/java/esprit_market/entity/marketplace/Product.java
package esprit_market.entity.marketplace;

import esprit_market.Enum.marketplaceEnum.ProductStatus;
import lombok.*;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Document(collection = "products")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    @Id
    private ObjectId id;
    private String name;
    private String description;

    private ObjectId shopId;

    @Builder.Default
    private List<ProductImage> images = new ArrayList<>();

    @Builder.Default
    private List<ObjectId> categoryIds = new ArrayList<>();

    private double price;
    private int stock;
    
    // ✅ Added missing fields
    private int quantity; // For compatibility with existing code
    
    @Builder.Default
    private ProductStatus status = ProductStatus.PENDING;
    
    // Approval workflow fields
    private LocalDateTime createdAt;
    private LocalDateTime approvedAt;
    private ObjectId approvedBy;
    private String rejectionReason;
}

File: frontend/src/app/front/models/product.ts
export interface Product {
    id: string;
    name: string;
    description: string;
    price: number;
    originalPrice?: number;
    category: string;
    categoryIds?: string[]; // ✅ Added for backend compatibility
    subcategory?: string;
    imageUrl: string;
    images?: string[];
    sellerId: string;
    shopId?: string; // ✅ Added shopId field for shop filtering
    sellerName: string;
    sellerAvatar?: string;
    rating: number;
    reviewsCount: number;
    stock: number;
    stockStatus: StockStatus;
    condition: ProductCondition;
    tags?: string[];
    isNegotiable: boolean;
    isFavorite?: boolean;
    viewCount?: number;
    createdAt?: Date;
    updatedAt?: Date;
    // New stock management fields
    maxPurchaseQuantity?: number;
    isAvailable?: boolean;
    stockWarningThreshold?: number;
    // ✅ APPROVAL WORKFLOW FIELDS
    status?: ProductStatus;
    approvedAt?: Date;
    approvedBy?: string;
    rejectionReason?: string;
    // ========================================
    // TRUST & REPUTATION SYSTEM FIELDS
    // ========================================
    /**
     * Seller's trust score (0-100)
     */
    trustScore?: number;
    /**
     * Seller's trust badge
     * Values: NEW_SELLER, GROWING_SELLER, TRUSTED_SELLER, TOP_SELLER
     */
    trustBadge?: string;
}

export enum ProductStatus {
    PENDING = 'PENDING',
    APPROVED = 'APPROVED',
    REJECTED = 'REJECTED',
    SUSPENDED = 'SUSPENDED'
}

export enum StockStatus {
    IN_STOCK = 'IN_STOCK',
    LOW_STOCK = 'LOW_STOCK', 
    OUT_OF_STOCK = 'OUT_OF_STOCK',
    DISCONTINUED = 'DISCONTINUED'
}

export enum ProductCondition {
    NEW = 'NEW',
    LIKE_NEW = 'LIKE_NEW',
    GOOD = 'GOOD',
    FAIR = 'FAIR',
    POOR = 'POOR'
}

export interface ProductCategory {
    id: string;
    name: string;
    slug: string;
    icon: string;
    description?: string;
    parentId?: string;
    children?: ProductCategory[];
    productCount: number;
    productIds?: string[];
}

export interface Shop {
    id: string;
    userId: string;
    name: string;
    slug: string;
    description?: string;
    logo?: string;
    banner?: string;
    rating: number;
    reviewsCount: number;
    productCount: number;
    totalProducts?: number;  // ✅ Added for backend compatibility
    totalSales: number;
    isVerified: boolean;
    isActive?: boolean;  // ✅ Added for backend compatibility
    joinedAt: Date;
    responseTime?: string;
    responseRate?: number;
    categories?: string[];
}

